slither/__main__.py

```diff
@@ -576,20 +586,46 @@ def main_impl(all_detector_classes, all_printer_classes):
             number_contracts = 0
             results = []
             if args.splitted and args.solc_ast:
-                (results, number_contracts) = process_files(filenames, args, detector_classes, printer_classes)
+                (slither, results, number_contracts) = process_files(filenames, args, detector_classes, printer_classes)
             else:
                 for filename in filenames:
-                    (results_tmp, number_contracts_tmp) = process(filename, args, detector_classes, printer_classes)
+                    (slither, results_tmp, number_contracts_tmp) = process(filename, args, detector_classes, printer_classes)
```

Comment:
the `slither` will not be correct here, as it is a loop.
The variable seems to be only used for the `compilation` output. I am not sure about keeping this output, as it's a duplicate of `crytic-compile` standard export format, without following the same json format.

I think we should either, we drop `compilation` here, or we create `crytic_compile.export_json`, that we store into a `compilations` list. Any thought?

Reply:
I wouldn't drop `compilation` entirely. External tooling such as VSCode might find it useful to have all of this information, and might not be able to compile all of the build configurations which `crytic-compile` can on its own. Additionally, running a separate compilation process outside of slither would eat up too much time. 

I'm fine with aggregating all the compilation information in this loop and using forwarding JSON output from `crytic-compile` into slither JSON output as you had mentioned, as long as we output the same data and don't dump to disk (as disk I/O would be slow).

Reply:
This PR has been updated to change `compilations` into an array of all the compilations which took place (a globbed filename will many compilations + slither instances to analyze each target). Each compilation uses crytic-compile's standard export format now.